source/validate.cpp

```diff
@@ -31,6 +31,10 @@
 #include <string.h>
 
 #include <vector>
+#include <iterator>
+#include <set>
+
+using std::vector;
```

Comment:
The using declarations should come after all #includes.  Otherwise you might mess up the type mappings seen by those other includes.